### PR TITLE
Webpack2

### LINKED ISSUE #4: Remove babel-polyfill dependency
Just an FYI that I will remove this dependency from the header component. Reasons:

* It is only used for the local server.
* Webpack uses it when it bundles the code, but we don't even use webpack to distribute. Currently we use `npm run babel-build` which doesn't use webpack.
* If we remove the `babel-polyfill` dependency, we can run`npm run build` to build the component which should be a smaller distributed file.
* This should not affect any app running on IE11 or other older browsers because each app uses `babel-polyfill` itself and that responsibility, I think, should be on the app.

If this makes sense to go ahead, then I think it's safe to remove and we can then have a smaller header dist folder.

CC @ktp242 @rhernand3z

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes #4 - removing babel-polyfill (but apps must still have it).

* Updating to webpack 2.
* Using webpack to minify the JS into one single `index.min.js` file instead of running `npm run babel-build`. Now it's `npm run dist` and consistent with our other apps/components.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Currently deployed to http://header-dev.us-east-1.elasticbeanstalk.com .

File: webpack.config.js [hunk 1]
```diff
@@ -29,37 +44,42 @@ if (process.env.NODE_ENV !== 'development') {
       },
     },
     module: {
-      loaders: [
```
Comment (rhernand3z) [inline webpack.config.js]: @EdwinGuzman 

If we add `react-dom` to the externals, it reduces the file size from 1.73mb to 1.06 which is huge.

```
      react: {
        root: 'React',
        commonjs2: 'react',
        commonjs: 'react',
        amd: 'react',
        umd: 'react',
      },
      'react-dom': {
        root: 'ReactDOM',
        commonjs2: 'react-dom',
        commonjs: 'react-dom',
        amd: 'react-dom',
        umd: 'react-dom',
      },
```

We could also add `underscore`

Comment (EdwinGuzman) [reply webpack.config.js]: Good catch. Will do.

**Comment by EdwinGuzman** [comment]
@rhernand3z yes, but that change would be for the header app, not the component. Are you sure it's in the travis.yml file or is it part of the package.json config?

**Comment by rhernand3z** [comment]
@EdwinGuzman -- When travis attempts to 1) npm install 2) npm run dist -- IMO it should mimick the node environment where the header component will be consumed and where the npm run test should be executed. Node 5 is pretty old now and I guess I want node/npm on travis to be consistent across our apps

**Comment by EdwinGuzman** [comment]
Cool. Updated the version. Let's see how the tests go.

**Pull request merged**